cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -1007,6 +1010,62 @@ func deleteAllPodsBoundToNode(ctx *controllerContext, nodeName string) error {
 	return utilerrors.NewAggregate(errs)
 }
 
+func markFailedAndDeletePodWithCondition(ctx *controllerContext, pod *v1.Pod) error {
+	const fieldManager = "GCPControllerManager"
+	condition := corev1apply.PodCondition().
+		WithType(v1.DisruptionTarget).
+		WithStatus(v1.ConditionTrue).
+		WithReason("DeletionByGCPControllerManager").
+		WithMessage(fmt.Sprintf("%s: node no longer exists", fieldManager)).
+		WithLastTransitionTime(metav1.Now())
+
+	// Based on pod gc controller (https://github.com/kubernetes/kubernetes/blob/17bf864c1fc20e4badc9a6e0659c5190310462e1/pkg/controller/podgc/gc_controller.go)
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+
+		// Extract the pod status as GCPControllerManager may or may not own the pod phase, if
+		// it owns the phase then we need to send the field back if the condition
+		// is added.
+		podApply, err := corev1apply.ExtractPodStatus(pod, fieldManager)
+		if err != nil {
+			return nil
+		}
+
+		// Set the status in case PodGC does not own any status fields yet
+		if podApply.Status == nil {
+			podApply.WithStatus(corev1apply.PodStatus())
+		}
+
+		// Mark the pod as failed - this is especially important in case the pod
+		// is orphaned, since there is no kubelet running anymore to change the phase.
+		if pod.Status.Phase != v1.PodSucceeded && pod.Status.Phase != v1.PodFailed {
+			// Only set the condition if the pod is not already in a terminal phase (Succeded or Failed).
+			updatePodCondition(podApply.Status, condition)
+			podApply.Status.WithPhase(v1.PodFailed)
+		}
+		if _, err := ctx.client.CoreV1().Pods(pod.Namespace).ApplyStatus(context.TODO(), podApply, metav1.ApplyOptions{FieldManager: fieldManager, Force: true}); err != nil {
+			return err
+		}
```

Comment:
this should be inside the conditional above.

Reply:
Done.